### DISCUSSION TITLE
大神们救命：treeTable在合并单元格后，节点展开显示异常！

### DISCUSSION THREAD

**Discussion by zzlatm** [layui/layui#1674] [问答]
### 议题条件

- [X] 我确认已查看官方使用文档：https://layui.dev ，但没有找到相关解决方案。
- [X] 我确认已在 [Issues](https://github.com/layui/layui/issues?q=is%3Aissue ) 中搜索过类似的问题，但没有找到相关解决方案。
- [X] 我已仔细阅读： 🍀[ Layui Issue 贡献指南](https://github.com/layui/layui/blob/main/CONTRIBUTING.md)

### 议题类型

疑是 BUG

### 使用版本

v2.9.6

### 问题描述

layui加载了第三方插件对表格单元格进行了合并操作，然后treeTable的节点展开后就显示异常，如下图：
<img width="577" alt="图一" src="https://github.com/layui/layui/assets/161801547/619f63c6-540b-4e5d-b576-9f5d106a1f73">
<img width="526" alt="图2" src="https://github.com/layui/layui/assets/161801547/18476624-70d3-4514-a33f-9a2fc9654fe4">

原因肯定是插件对表格进行合并单无格后，父行就的单元格就变少了，所以子行一旦展开就显示异常了，大神们请问如何处理？

### 业务代码

```auto
var options={
  elem   :'#myTable'
  ,url    :''
  //
   //
  ,done :function(obj,curr,count){			
    layer.close(load);
    tableMerge.render(this);  //
  }
}
```

### 截图补充

<img width="577" alt="图一" src="https://github.com/layui/layui/assets/161801547/0d548be5-2924-44c4-a481-4c1ed14533ae">
<img width="526" alt="图2" src="https://github.com/layui/layui/assets/161801547/84ce7d01-2bb6-4211-9ef3-41d1fd5ed786">

### 浏览器

EDGE

### 友好承诺

- [X] 我承诺将本着相互尊重、理解和友善的态度进行交流，共同维护 Layui 良好的社区氛围。

**Comment by Sight-wcg** [comment]
试试展开/折叠后重新合并单元格

**Comment by zzlatm** [comment]
> 试试展开/折叠后重新合并单元格

不行的

**Reply by lvwenju** [reply]
试试从后往前合并，例如从最后的列最后的行往前合并。我没用插件，自己写的func，就这么实现的。

    function mergeCell(tableLayId, startRow=3, endRow=100, col) {
        let table = $('div[lay-id="' + tableLayId + '"]'),
            rowspan,
            currentCell;
        for (let r = startRow; r < endRow; r += rowspan) {
            currentCell = table.find('tr').eq(r).find('td').eq(col);
            let firstColumnCell = currentCell.siblings().eq(0);
            rowspan = 1;
            table.find('tr').eq(r).nextUntil(endRow).each(function () {
                let next = $(this).find('td').eq(col);
                let nextFirstColumnCell = next.siblings().eq(0);
                if (col === 0) {
                    if (currentCell.text() === next.text()) {
                        rowspan++;
                        next.remove();
                    }
                } else {
                    if (currentCell.text() === next.text() && firstColumnCell.text() === nextFirstColumnCell.text()) {
                        rowspan++;
                        next.remove();
                    }
                }
            });
            currentCell.attr('rowspan', rowspan);
        }
    }
